### ISSUE TITLE
"number" validation fails on string with valid basic float contrary to documentation

### ISSUE THREAD

**Issue by e3b0c442** [go-playground/validator#940]
- [x] I have looked at the documentation [here](https://pkg.go.dev/github.com/go-playground/validator/v10#section-documentation) first?
- [x] I have looked at the examples provided that may showcase my question [here](/_examples)?

### Package version eg. v9, v10: 
v10

### Issue, Question or Enhancement:
Per the documentation, the `number` validator _"validates that a string value contains number values only. *For integers or float it returns true.* [ref](https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Number)"_ 

The validation fails on a string that is a basic floating point number, _e.g._ `1.618`.

The [regex](https://github.com/go-playground/validator/blob/v10.11.0/regexes.go#L11) does not have any accommodation for the decimal point, so floats provided as string will fail the test. The validation function itself tries to coerce the type to a number type before using the regex, so values that are actually a `float` type will pass. Additionally but not in the scope of this bug, based on that regex, `number` will not handle negative numbers in a string, whether integral or floating point.

Possible correction courses:
1. Update the documentation to clarify that for values provided as a string, any non-digit character will fail the expression
2. Update the regex to handle basic floating point and negative numbers 
3. Update the function to attempt to parse the string as a number such that all cases in which a native numeric type would pass, the string would also pass

After exploring the code a bit, option 1 makes the most sense to me as it seems like the "numeric" validator handles the case I'm trying to achieve, while the documentation is just unclear.

### Code sample, to showcase or reproduce:

```go
	intString := "1"
	fmt.Printf("error with int: \"%v\"\n", validate.Var(intString, "number"))
	floatString := "1.618"
	fmt.Printf("error with float: \"%v\"\n", validate.Var(floatString, "number"))
```

Output:
```
error with int: "<nil>"
error with float: "Key: '' Error:Field validation for '' failed on the 'number' tag"
```

**Comment by e3b0c442** [comment]
After reading the docs multiple times I think I understand where the confusion is. Suggest rewording that documentation as follows:

`validates that a string-typed value contains digits only. For integer- or float-typed values, it returns true.`

**Comment by philippgille** [comment]
Instead of the documentation improvement, can't the [regex](https://github.com/go-playground/validator/blob/8f07b0368280d9fc70b4ffd3708b31427bbc286d/regexes.go#L11) be changed to validate floats? Or is that not very straightforward, like the discussion [on SO](https://stackoverflow.com/questions/12643009/regular-expression-for-floating-point-numbers) points out?
